### PR TITLE
javier_ruiz

### PR DESCRIPTION
**Pull request by japi-kolokium**
Solutions to the exercises for the Kubernetes CKAD Training - Session 1

### CODE REVIEW

File: javier_ruiz/session-01/exercise-01/command.bash [hunk 1]
```diff
@@ -0,0 +1,4 @@
+kubectl apply -f exercise-01-namespace.yaml
+kubectl apply -f .
+kubectl config set-context $(k config current-context) --namespace=exercise-01
+watch  'kubectl get pod -l app=wordpress -o name | while read pod; do echo; echo --- $pod; echo; kubectl logs $pod|tail -15; done'
```
Comment (juan131) [inline javier_ruiz/session-01/exercise-01/command.bash]: This last command is not working very well on my terminal

File: javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+apiVersion: extensions/v1beta1
```
Comment (juan131) [inline javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml]: No canaries on exercise2!!!

File: javier_ruiz/session-01/exercise-02/wordpress-service.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  namespace: exercise-02
+spec:
+  type: LoadBalancer
+  ports:
+  - name: wordpress
+    port: 8001
+    targetPort: 80
+  selector:
+    app: wordpress
+  sessionAffinity: ClientIP
```
Comment (juan131) [inline javier_ruiz/session-01/exercise-02/wordpress-service.yaml]: Not needed once there's no canary

File: javier_ruiz/session-01/exercise-01/command.bash [hunk 4]
```diff
@@ -0,0 +1,4 @@
+kubectl apply -f exercise-01-namespace.yaml
+kubectl apply -f .
+kubectl config set-context $(k config current-context) --namespace=exercise-01
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-01/command.bash]: FIx

File: javier_ruiz/session-01/exercise-01/exercise-01-namespace.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-01/exercise-01-namespace.yaml]: None. I suppose it was a test that I didn't clean.
Fixed.

File: javier_ruiz/session-01/exercise-01/wordpress-service.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress
+  namespace: exercise-01
+spec:
+  type: LoadBalancer
+  ports:
+  - name: wordpress
+    port: 8001
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-01/wordpress-service.yaml]: I was testing the exercise in my local kubernetes enviroment and I never now if I'm already using the 80 port in another labs. Changed to 80 and added 443.

File: javier_ruiz/session-01/exercise-02/command.bash [hunk 7]
```diff
@@ -0,0 +1,4 @@
+kubectl apply -f exercise-02-namespace.yaml
+kubectl apply -f .
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-02/command.bash]: Didn't see the need to create them in order. Need more testing but kubernetes seems to handle the order of creation ok in this exercise except for the namespace.

File: javier_ruiz/session-01/exercise-02/mariadb-master-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+              key: password
+        - name: MARIADB_REPLICATION_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb
+              key: replication_password
+        ports:
+        - name: mariadb-master
+          containerPort: 3306
+        resources:
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-02/mariadb-master-deployment.yaml]: Jeje .. Yes, it's clear that I have mainly worked in Infrastructure. I wanted to know what do you think about setting limits so low. I'll double them.

File: javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        env:
+        - name: WORDPRESS_DATABASE_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: wordpress
+              key: database.password
+        ports:
+        - name: wordpress
+          containerPort: 80
+        resources:
```
[the remark this answers is not in the document]
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml]: Doubled

File: javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml (hunk 2 above)
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-02/wordpress-canary-deployment.yaml]: Deleted wordpress-canary deployment file

File: javier_ruiz/session-01/exercise-02/wordpress-service.yaml (hunk 3 above)
Comment (japi-kolokium) [reply javier_ruiz/session-01/exercise-02/wordpress-service.yaml]: Updated.

File: javier_ruiz/session-01/exercise-01/command.bash [hunk 10]
```diff
@@ -1,4 +1,4 @@
 kubectl apply -f exercise-01-namespace.yaml
 kubectl apply -f .
-kubectl config set-context $(k config current-context) --namespace=exercise-01
+kubectl config set-context $(kubectl config current-context) --namespace=exercise-01
```
Comment (juan131) [inline javier_ruiz/session-01/exercise-01/command.bash]: better! 😉

File: javier_ruiz/session-01/exercise-01/wordpress-canary-deployment.yaml [hunk 11]
```diff
@@ -26,21 +26,38 @@ spec:
         - configMapRef:
             name: wordpress
         env:
```
Comment (juan131) [inline javier_ruiz/session-01/exercise-01/wordpress-canary-deployment.yaml]: Better setup!

**Pull request merged**